tasks/configuration.yml

```diff
@@ -17,22 +17,48 @@
 # limitations under the License.
 #
 
-- name: Configure Horizon
-  lineinfile: dest={{ dashboard_settings }}
-              regexp=^OPENSTACK_HOST
-              line="OPENSTACK_HOST = '{{ keystone_hostname }}'"
-  notify:
-    - Restart Apache2
+- name: Configure openstack-dashboard local_settings file
+  lineinfile:
+    dest: '{{ dashboard_settings }}'
+    regexp: '{{ item.regexp }}'
+    line: '{{ item.line }}'
+  with_items:
+    - {regexp: '^OPENSTACK_HOST =', line: "OPENSTACK_HOST = '{{ keystone_hostname }}'"}
+    - {regexp: '^ALLOWED_HOSTS =', line: "ALLOWED_HOSTS = ['*', ]"}
+    - {regexp: '^SESSION_ENGINE =', line: "SESSION_ENGINE = 'django.contrib.sessions.backends.cache'"}
+    - {regexp: '^OPENSTACK_KEYSTONE_URL =', line: "OPENSTACK_KEYSTONE_URL = 'http://%s:5000/v3' % OPENSTACK_HOST"}
+    - {regexp: '^OPENSTACK_KEYSTONE_MULTIDOMAIN_SUPPORT =', line: 'OPENSTACK_KEYSTONE_MULTIDOMAIN_SUPPORT = True'}
+    - {regexp: '^OPENSTACK_KEYSTONE_DEFAULT_DOMAIN =', line: "OPENSTACK_KEYSTONE_DEFAULT_DOMAIN = 'default'"}
+    - {regexp: '^OPENSTACK_KEYSTONE_DEFAULT_ROLE =', line: "OPENSTACK_KEYSTONE_DEFAULT_ROLE = 'user'"}
 
-- name: Allow all hosts
-  lineinfile: dest={{ dashboard_settings }}
-              regexp=^ALLOWED_HOSTS
-              line="ALLOWED_HOSTS = ['*']"
-  when: ansible_os_family == 'RedHat'
-  notify:
-    - Restart Apache2
+- name: Remove default cache configuration
+  replace:
+    dest: '{{ dashboard_settings }}'
+    regexp: "CACHES = {\n    'default': {\n        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',\n    \
+      },\n}"
+    replace: ''
 
-- name: Extend horizon session timeout
-  lineinfile: dest={{ dashboard_settings }} line="SESSION_TIMEOUT=604800"
-  notify:
-    - Restart Apache2
+- name: Configuare memcached session storage service
+  blockinfile:
+    dest: '{{ dashboard_settings }}'
+    marker: '# Memcached session storage'
+    block: |
+      CACHES = {
+          'default': {
+              'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
+              'LOCATION': 'localhost:11211',
+          }
+      }
+  changed_when: False
```

Comment:
I proposed that, because of the number, the nature and size of the changes. 
But as @sibradzic mentioned, could be a problem versioning every template for every OSt release, as the base template would change a lot across them.